console/console-server/src/main/resources/iot.schema.graphql

```diff
@@ -0,0 +1,231 @@
+enum ProjectPhaseType {
+  Active
+  Configuring
+  Terminating
+  Failed
+}
+
+enum iotProjectType {
+  managed
+  external
+}
+
+enum iotCredentials {
+  psk
+  hashed_password
+  x509
+}
+
+union iotProjectDownStreamStrategy =
+  externalDownstreamStrategy_iot_enmasse_io_v1alpha1 | managedDownstreamStrategy_iot_enmasse_io_v1alpha1
+
+type credentials_iot_enmasse_io_v1alpha1{
+  username: String!
+  password: String!
+}
+
+type connectionInformation_iot_enmasse_io_v1alpha1{
+  host: String!
+  port: Int!
+  credentials: credentials_iot_enmasse_io_v1alpha1!
+
+  tls : Boolean!
+  certificate: String
+}
+
+type managedStatus_iot_enmasse_io_v1alpha1 {
+  passwordTime: Date
+  addressSpace: String
+}
+
+type IoTProjectStatus_iot_enmasse_io_v1alpha1 {
+  phase: ProjectPhaseType!
+  phaseReason: String
+  tenantName: String!
+  downstreamEndpoint: connectionInformation_iot_enmasse_io_v1alpha1!
+
+  managedStatus: managedStatus_iot_enmasse_io_v1alpha1
+}
+
+type AddressSpaceConfig_iot_enmasse_io_v1alpha1  {
+  name: String!
+  plan: String!
+  type: String!
+}
+
+type AddressConfig_iot_enmasse_io_v1alpha1  {
+  plan: String!
+  type: String!
+}
+
+type AddressesConfig_iot_enmasse_io_v1alpha1  {
+  Telemetry: AddressConfig_iot_enmasse_io_v1alpha1!
+  Event: AddressConfig_iot_enmasse_io_v1alpha1!
+  Command: AddressConfig_iot_enmasse_io_v1alpha1!
+}
+
+
+type managedDownstreamStrategy_iot_enmasse_io_v1alpha1 {
+  addressSpace: AddressSpaceConfig_iot_enmasse_io_v1alpha1!
+  addresses: AddressesConfig_iot_enmasse_io_v1alpha1!
+}
+
+type externalDownstreamStrategy_iot_enmasse_io_v1alpha1 {
+  connectionInformation: connectionInformation_iot_enmasse_io_v1alpha1!
+}
+
+type IotProjectSpec_iot_enmasse_io_v1alpha1 {
+  downstreamStrategy: iotProjectDownStreamStrategy !
+  configuration: String!
+}
+
+type IoTProject_iot_enmasse_io_v1alpha1 {
+
+  metadata: ObjectMeta_v1! @goField(name: "ObjectMeta")
+  enabled: Boolean!
+  spec: IotProjectSpec_iot_enmasse_io_v1alpha1!
+  status: IoTProjectStatus_iot_enmasse_io_v1alpha1!
+  devices(iotproject: String!, first: Int, offset: Int, filter: String, orderBy: String): devicesQueryResult_consoleapi_iot_enmasse_io_v1alpha1
+  adapters: [adapter]
+}
+
+interface adapter {}
+
+# TODO : map to k8s objectcs and pull endpoints details from openshift routes ?
+type HTTP_Adapter implements adapter {
+  url : String!
+}
+
+type MQTT_Adapter implements adapter {
+ tls_enabled: Boolean!
```

Comment:
is the switch to the _ separator style intentional?   Did you see https://www.apollographql.com/docs/apollo-server/schema/schema/#naming-conventions

Reply:
I think i got confused between the underscore you used to made the types closer to the kubernetes API. I redo a pass to align that.